pkg/controller/vsphere/reconciler.go

```diff
@@ -217,12 +232,21 @@ func (r *Reconciler) delete() error {
 		}
 		if moTask != nil {
 			if taskIsFinished, err := taskIsFinished(moTask); err != nil {
-				metrics.RegisterFailedInstanceDelete(&metrics.MachineLabels{
-					Name:      r.machine.Name,
-					Namespace: r.machine.Namespace,
-					Reason:    "Task finished with error",
-				})
-				return err
+				// Check if latest task is not a task for vm cloning
+				if taskIsFinished && moTask.Info.DescriptionId != cloneVmTaskDescriptionId {
+					metrics.RegisterFailedInstanceDelete(&metrics.MachineLabels{
+						Name:      r.machine.Name,
+						Namespace: r.machine.Namespace,
+						Reason:    "Task finished with error",
+					})
+					klog.Errorf("Delete task finished with error: %w", err)
+					return err
+				} else {
+					klog.Warningf(
+						"TaskRef points to clone task which finished with error: %w Proceeding with machine deletion.", err,
```

Comment:
I would expect that the error string is not punctuated, so we need to add that ourselves otherwise the last part of this message will look like it is part of the error
```suggestion
						"TaskRef points to clone task which finished with error: %w. Proceeding with machine deletion", err,
```